src/pages/_app.tsx

```diff
@@ -31,6 +33,14 @@ function MyApp({ Component, pageProps }) {
         {page}
       </Layout>
     ));
+
+  const router = useRouter();
+  let canonicalUrl = 'https://docs.amplify.aws';
+  const canonicalPath = CANONICAL_URLS.includes(router.pathname)
+    ? router.pathname.replace('[platform]', 'javascript')
+    : router.asPath;
```

Comment:
If the current `router.pathname` doesn't exist in the list, then we just use the `router.asPath` as the canonical URL. `router.asPath` will be the url that includes the current platform